llama-index-integrations/readers/llama-index-readers-toggl/llama_index/readers/toggl/base.py

```diff
@@ -0,0 +1,80 @@
+import asyncio
+import datetime
+from typing import List, Optional
+
+from llama_index.core.readers.base import BaseReader
+from llama_index.core.schema import Document
+
+from llama_index.readers.toggl.dto import TogglTrackItem, TogglOutFormat
+
+
+class TogglReader(BaseReader):
+    def __init__(
+        self, api_token: str, user_agent: str = "llama_index_toggl_reader"
+    ) -> None:
+        """Initialize with parameters."""
+        super().__init__()
+        self.api_token = api_token
+        self.user_agent = user_agent
+        self.loop = asyncio.new_event_loop()
```

Comment:
I'm not 100% sure this will work in a notebook? But we can always update later too